### PR TITLE
bulk-crap-uninstaller: fix scripts

### LINKED ISSUE #4503: bulk-crap-uninstaller: scripts not work properly
Scripts in `uninstaller` filed just creates a blank file named `bulk-crap-uninstaller` instead of copying settings file into persist folder,
cause there is no folder for the app in persist directory.
#4499 solution here, please review

### PR DESCRIPTION
**Pull request by AkariiinMKII**
- Closes #4503 

Fix the problem that installer and uninstaller scripts not work properly, and persist settings file.

### CODE REVIEW

File: bucket/bulk-crap-uninstaller.json [hunk 1]
```diff
@@ -23,6 +19,24 @@
...
+    ],
+    "uninstaller": {
+        "script": [
+            "if (!(Test-Path \"$persist_dir\\BCUninstaller.settings\")) {",
+            "    if (Test-Path \"$dir\\BCUninstaller.settings\") {",
+            "        Copy-Item \"$dir\\BCUninstaller.settings\" \"$persist_dir\" -Force",
+            "    }",
+            "}"
+        ]
+    },
```
Comment (Ash258) [inline bucket/bulk-crap-uninstaller.json:39]:
```suggestion
```

File: bucket/bulk-crap-uninstaller.json [hunk 2]
```diff
@@ -5,10 +5,6 @@
     "license": "Apache-2.0",
     "url": "https://github.com/Klocman/Bulk-Crap-Uninstaller/releases/download/v4.16/BCUninstaller_4.16_portable.zip",
     "hash": "d383996cc87a82bbb6c51f775d711c887db6e44687d848b6fb0792fde703d548",
-    "pre_install": "Copy-Item \"$persist_dir\\BCUninstaller.settings\" \"$dir\" -ErrorAction SilentlyContinue -Force",
-    "uninstaller": {
-        "script": "Copy-Item \"$dir\\BCUninstaller.settings\" \"$persist_dir\" -ErrorAction SilentlyContinue -Force"
-    },
```
Comment (Ash258) [inline bucket/bulk-crap-uninstaller.json:11]:
```suggestion
    "pre_install": [
        "ensure \"$persist_dir\" | Out-Null",
        "Copy-Item \"$persist_dir\\BCUninstaller.settings\" \"$dir\" -ErrorAction SilentlyContinue"
    ],
    "uninstaller": {
        "script": "Copy-Item \"$dir\\BCUninstaller.settings\" \"$persist_dir\" -ErrorAction SilentlyContinue -Force"
    },
```

Comment (AkariiinMKII) [reply bucket/bulk-crap-uninstaller.json:11]: According to my test result, `BCUninstaller.settings` was edited while exiting the app, not overwrited, and the file link works well in this situation.

Comment (Ash258) [reply bucket/bulk-crap-uninstaller.json:11]: Not it is not. FIrst start of installed application will delete the file, breaking link and create new file.

image

First startup:

File is deleted as soon as application started

A

Comment (AkariiinMKII) [reply bucket/bulk-crap-uninstaller.json:11]: Please check if there is any error in setting file, or it is just blank, the app only delete setting file while it is not loaded properly.

**Pull request closed without merging**